### PR TITLE
[CAT-1067] Initial version of onfido-java generated from onfido-openapi-spec

### PR DESCRIPTION
**Pull request by devops-automation-onfido**
- Auto-generated PR

### CODE REVIEW

**Comment by dvacca-onfido** [comment]
The [first commit](https://github.com/onfido/onfido-java/pull/101/commits/757186e7d6a0ced6c54a32dd51cfbdf5bf242b13) has been automatically generated from [this](https://github.com/onfido/onfido-openapi-spec/tree/bc1ac7519d3fb4370c2470266031a61dcace417f) version of onfido-openapi-spec using [this](https://github.com/onfido/onfido-openapi-spec/blob/bc1ac7519d3fb4370c2470266031a61dcace417f/generators/java/config.yaml) configuration on top of common [one](https://github.com/onfido/onfido-openapi-spec/blob/bc1ac7519d3fb4370c2470266031a61dcace417f/generators/common/config.yaml).

Afterwards previous tests has been moved or dropped [here](https://github.com/onfido/onfido-java/pull/101/commits/f3e460bea62bfc766b4288175248e1f56ece9150), in a separate commit to allow tracking the changes. 

Subsequent [change](https://github.com/onfido/onfido-java/pull/101/commits/a5ab329bdca0b75c84fe67f4eeda61abdef46bda) has been done by hand to adapt tests to the new library.

Linter has been then executed in [this](https://github.com/onfido/onfido-java/pull/101/commits/2abe12b2fbd578a9f65f4917375fdd73af065635) commit and some [next commits](https://github.com/onfido/onfido-java/pull/101/files/2abe12b2fbd578a9f65f4917375fdd73af065635..5bcae3a581cf1d6c380cf82bc418157f0bde0895) have been done again by hand to make tests working well in CI. 

In a nutshell, changes that needs carefully attentions are below:
- [Adaptations of the tests](https://github.com/onfido/onfido-java/pull/101/commits/a5ab329bdca0b75c84fe67f4eeda61abdef46bda)
- [Fixes on CI](https://github.com/onfido/onfido-java/pull/101/files/2abe12b2fbd578a9f65f4917375fdd73af065635..5bcae3a581cf1d6c380cf82bc418157f0bde0895)

@DavidMealha-Onfido @RiccardoPetteruti-Onfido

File: .maven.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
```
Comment (dvacca-onfido) [inline .maven.xml:1]: This change has been pushed back from the last version [here](https://github.com/onfido/onfido-java/blob/master/.maven.xml) since needed by the workflow to publish the package. It's going to be ignored from the files that are automatically overridden on library generation.

File: src/test/java/com/onfido/integration/ReportTest.java [hunk 2]
```diff
@@ -1,91 +1,141 @@
...
+    try {
+      return report.getDocumentReport().getId();
+    } catch ( Exception ClassCastException )
+    {
+      return report.getIdentityEnhancedReport().getId();
+    }
   }
 
-  @Test
+  // @Test
```
[the remark this answers is not in the document]
Comment (dvacca-onfido) [reply src/test/java/com/onfido/integration/ReportTest.java:64]: Good spot. No, it was forgotten.

[the remark this answers is not in the document]
Comment (dvacca-onfido) [reply src/test/java/com/onfido/integration/ReportTest.java]: Fixed [here](https://github.com/onfido/onfido-java/pull/101/commits/cf2d9144fb3a889acffa48f0ce90f5c535ba05f1)

**Review by RiccardoPetteruti-Onfido — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#
```
Comment (DavidMealha-Onfido) [inline .travis.yml:1]: Should this be ignored?

Comment (dvacca-onfido) [reply .travis.yml:1]: Yeah, as we don't use travis CI it's pointless committing it.

**Review by DavidMealha-Onfido — APPROVED** [review]
